app/components/forms/events/view/edit-access-code.js

```diff
@@ -0,0 +1,151 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import $ from 'jquery';
+import FormMixin from 'open-event-frontend/mixins/form';
+
+export default Component.extend(FormMixin, {
+  getValidationRules() {
+    $.fn.form.settings.rules.checkMaxMin = () => {
+      return this.get('data.minQuantity') <= this.get('data.maxQuantity');
+    };
+    $.fn.form.settings.rules.checkMaxTotal = () => {
+      return this.get('data.maxQuantity') <= this.get('data.ticketsNumber');
+    };
+    return {
+      inline : true,
+      delay  : false,
+      on     : 'blur',
+
+      fields: {
+        accessCode: {
+          identifier : 'access_code',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.get('l10n').t('Please enter access code')
+            },
+            {
+              type  : 'regExp',
+              value : '^[a-zA-Z0-9_-]*$'
```

Comment:
@pradeepgangwar but all the regular expressions are in utils/validators.js file and then imported.